An unhandled exception occurred: Cannot find module 'nx/src/config/workspaces' since v14.0.1

<!--
Please don't ignore this template.

If you ignore it, we're just going to respond asking you to fill it out, which wastes everyone's time.
The more relevant information you can include, the faster we can find the issue and fix it without asking you for more info.
-->

<!--
🚨 STOP 🚨 𝗦𝗧𝗢𝗣 🚨 𝑺𝑻𝑶𝑷 🚨

This issue template is only for problems specifically with the `@angular-eslint/builder` package.
That means you should only use this template if there are problems running ESLint on your workspace.

If you have a problem with a specific lint rule, please back out and select the `@angular-eslint/eslint-plugin` or `@angular-eslint/eslint-plugin-template` issue template, depending on where the rule lives.
-->

**Description and reproduction of the issue**

See https://github.com/jnizet/decouvertes-nature/pull/284:

- checkout the branch of the PR (renovate/angular-eslint-monorepo)
- run `yarn`
- run `yarn lint`

It passed with version 14.0.0. With version 14.0.1, it fails with the following error:

```
An unhandled exception occurred: Cannot find module 'nx/src/config/workspaces'
Require stack:
- /home/runner/work/decouvertes-nature/decouvertes-nature/node_modules/@nrwl/devkit/index.js
- /home/runner/work/decouvertes-nature/decouvertes-nature/node_modules/@angular-eslint/builder/dist/compat.js
- /home/runner/work/decouvertes-nature/decouvertes-nature/node_modules/@angular-devkit/architect/node/node-modules-architect-host.js
- /home/runner/work/decouvertes-nature/decouvertes-nature/node_modules/@angular-devkit/architect/node/index.js
- /home/runner/work/decouvertes-nature/decouvertes-nature/node_modules/@angular/cli/src/command-builder/architect-base-command-module.js
- /home/runner/work/decouvertes-nature/decouvertes-nature/node_modules/@angular/cli/src/command-builder/architect-command-module.js
- /home/runner/work/decouvertes-nature/decouvertes-nature/node_modules/@angular/cli/src/commands/build/cli.js
- /home/runner/work/decouvertes-nature/decouvertes-nature/node_modules/@angular/cli/src/command-builder/command-runner.js
- /home/runner/work/decouvertes-nature/decouvertes-nature/node_modules/@angular/cli/lib/cli/index.js
- /home/runner/work/decouvertes-nature/decouvertes-nature/node_modules/@angular/cli/lib/init.js
```

<!--
Please consider creating an isolated reproduction repo to make it easy for the volunteer maintainers debug your issue.
-->

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@angular-eslint/builder`          | `14.0.1` |
| `ESLint`                           | `8.19.0` |
| `node`                             | `14.19.3` |

```sh
# Please run `npx ng version` in your project and paste the full output here:

Angular CLI: 14.0.5
Node: 14.19.3
Package Manager: yarn 1.22.19 
OS: darwin x64

Angular: 14.0.5
... animations, cli, common, compiler, compiler-cli, core, forms
... localize, platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1400.5
@angular-devkit/build-angular   14.0.5
@angular-devkit/core            14.0.5
@angular-devkit/schematics      14.0.5
@angular/fire                   7.4.1
@schematics/angular             14.0.5
rxjs                            7.5.5
typescript                      4.7.4
```

<!--
^ **Before submitting the issue** please check that output from `ng version` carefully.

Is there any inconsistency between your major version numbers? E.g. if you see `@angular-devkit/core` at version `7.x.x` but your `@angular/cli` is a version `11.x.x` it makes sense that you would be experiencing issues.

It's not clear how workspaces can end up in this state, but fixing it is just a case of installing the correct versions that are intended to work together and verifying the `ng version` output once again.
-->

- [x] I have updated to the latest supported version of the packages and checked my `ng version` output per the instructions given here.

--- a/packages/builder/package.json
+++ b/packages/builder/package.json
@@ -19,7 +19,8 @@
   ],
   "builders": "./builders.json",
   "dependencies": {
-    "@nrwl/devkit": "^14.2.4"
+    "@nrwl/devkit": "^14.2.4",
+    "nx": "^14.2.4"
   },
   "peerDependencies": {
     "eslint": "^7.0.0 || ^8.0.0",

